src/crypto/slow-hash.c

```diff
@@ -1070,6 +1069,13 @@ void cn_slow_hash(const void *data, size_t length, char *hash)
         hash_extra_blake, hash_extra_groestl, hash_extra_jh, hash_extra_skein
     };
 
+#ifndef FORCE_USE_HEAP
+    uint8_t long_state[MEMORY];
+#else
+    uint8_t *long_state = NULL;
+    long_state = (uint8_t *)malloc(MEMORY);
+#endif
```

Comment:
Can you use `unique_ptr` then? No need for the second `#ifndef` below, and do not have to be concerned about throwing functions in here.. Make sure to not mix `new`/`free` or `new[]`/`delete`. Easiest is to do `unique_ptr<char[]> block{new char[MEMORY]}`.